services/src/main/java/org/keycloak/authentication/forms/RegistrationRecaptcha.java

```diff
@@ -80,7 +80,9 @@ protected boolean validate(ValidationContext context, String captcha, Map<String
         List<NameValuePair> formparams = new LinkedList<>();
         formparams.add(new BasicNameValuePair("secret", config.get(SECRET_KEY)));
         formparams.add(new BasicNameValuePair("response", captcha));
-        formparams.add(new BasicNameValuePair("remoteip", context.getConnection().getRemoteAddr()));
+        if (context.getConnection().getRemoteAddr() != null) {
+            formparams.add(new BasicNameValuePair("remoteip", context.getConnection().getRemoteAddr()));
```

Comment:
Do we know the implications on the form if the IP is missing?

Reply:
For both of these scnenarios the ip is an optional parameter. It seems to be used as an anti-fraud mechanism. If it's missing my guess is that you're more likely to see throttling.

Reply:
Ok, makes sense. Thanks.